File: Algorithms/SelectionSort/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    static int[] selectionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int endIdx = arr.length - 1 - i;
            int maxNumIdx = getMaxNumIdx(arr, 0, endIdx);

            // Swap the nums by their index
            int temp = arr[maxNumIdx];
            arr[maxNumIdx] = arr[endIdx];
            arr[endIdx] = temp;
        }

        return arr;

    }

    static int getMaxNumIdx(int[] arr, int startIdx, int endIdx) {
        int currMaxNumIdx = endIdx;

        for (int i = startIdx; i < endIdx; i++) {
            int currNum = arr[i];
            int currMaxNum = arr[currMaxNumIdx];

            if (currNum > currMaxNum) {
                currMaxNumIdx = i;
            }
        }

        return currMaxNumIdx;
    }

    public static void main(String[] args) {
        int[] arr = new int[] { 5, 6, 4, 3, 8, 1 };
        int[] sortedArr = selectionSort(arr);
        System.out.println(Arrays.toString(sortedArr));
    }
}

File: Algorithms/SelectionSort/SelectionSort.py
def selection_sort(arr):
    for i in range(len(arr)):
        end_idx = len(arr) - 1 - i
        max_num_idx = get_max_num_idx(arr, 0, end_idx)

        arr[max_num_idx], arr[end_idx] = arr[end_idx], arr[max_num_idx]

    return arr


def get_max_num_idx(arr, start_idx, end_idx):
    curr_max_idx = end_idx

    for i in range(start_idx, end_idx):
        if arr[i] > arr[curr_max_idx]:
            curr_max_idx = i

    return curr_max_idx


if __name__ == "__main__":
    arr = [5, 4, 6, 7, 1, 2]
    print(selection_sort(arr))
